File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 6/25/2026, 8:40:01 AM
import sys
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = -(sys.maxsize)
        currentSum = 0
        for i in range(len(nums)):
            """
            Kadane's Algorithm
            """
            firstOption = nums[i]
            secondOption = nums[i] + currentSum
            currentSum = max(firstOption, secondOption)
            maxSum = max(maxSum, currentSum)
        return maxSum
            
            

File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 6/25/2026, 8:39:57 AM
class Solution {
    public int maxSubArray(int[] nums) {
        int curr = 0;
        int max = Integer.MIN_VALUE;
        for(int i = 0; i < nums.length; i++){
            curr = Math.max(curr, 0) + nums[i];
            max = Math.max(max, curr);
        }
        return max;
    }
}